components/tailscale/main.tf

```
@@ -0,0 +1,25 @@
+module "tailscale_subnet_router" {
+  source  = "git::https://github.com/masterpointio/terraform-aws-tailscale.git?ref=tags/0.1.0"
+  context = module.this.context
+
+  name = "tailscale-subnet-router"
+
+  vpc_id                  = local.vpc_id
+  advertise_routes        = [local.vpc_cidr_block]
+  subnet_ids              = local.private_subnet_ids
+  key_pair_name           = coalesce(var.tailscale_existing_ssh_key_name, module.ssh_key_pair.key_name)
+  session_logging_enabled = var.tailscale_session_logging_enabled
+}
+
+module "ssh_key_pair" {
```

```suggestion
module "ec2_key_pair" {
```

to differentiate that we aren't just creating a public/private key with something like [tls_private_key](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/private_key)